### ISSUE TITLE
Cluster Autoscaler Release 1.23

### ISSUE THREAD

**Issue by MaciekPytel** [kubernetes/autoscaler#4549]
I'm working on releasing CA 1.23 (including bumping Kubernetes dependencies, building images, etc). I'd like to ask cloudprovider owners to:
 * Please merge any last minute changes required. Please tag on this issue any changes to CA core that should be included in 1.23.0.
 * If you want to redact release notes for your cloudprovider please write them down in this issue and I will include them in the release notes for CA. If no release notes are posted I'll try to redact it based on the commits, but I obviously have limited context and I'd appreciate your help.

@ringtail @Jeffwan @jaypipes @gjtempleton @feiskyer @nilo19 @marwanad @hello2mao @davidjumani @enxebre @elmiko @hardikdr @detiber @andrewsykim @MorrisLaw @kevin-wangzefeng @RainbowMango @avorima @ellistarn @tghartland @d-mo @detiber @deitch @displague @gianarb

**Comment by marwanad** [comment]
Would like to get #4550 and #4441 in before 1.23 and both requiring top-level approvers.

**Comment by elmiko** [comment]
thanks @MaciekPytel 

this is the main release note for Cluster API provider

```
The `CAPI_GROUP` environment variable will now also affect the annotations keys for minimum and maximum node group size, the machine annotation, machine deletion, and the cluster name label. See PR #4451 for more details.
```

**Comment by MaciekPytel** [comment]
I've created cluster-autoscaler-release-1.23 branch. Any new PRs merged to master will need to be cherry-picked to be included in 1.23.

All PRs requested by @marwanad have been merged before the branch was cut and are included.

**Comment by MaciekPytel** [comment]
CA 1.23 has been released. Please let me know if I missed something or made any errors in release notes and I will fix them.
